client/state/support/actions.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SUPPORT_USER_ACTIVATED,
+	SUPPORT_USER_DEACTIVATED,
+	SUPPORT_USER_FETCH_TOKEN,
+} from 'state/action-types';
+
+import User from 'lib/user';
+
+import userSettings from 'lib/user-settings';
+
+/**
+ * Requests a support user token, then dispatches the relevant actions upon response
+ * 
+ * @param  {string} supportUser     Support username
+ * @param  {string} supportPassword Support password
+ * @return {thunk}                  The action thunk
+ */
+export function supportUserFetchToken( supportUser, supportPassword ) {
+	return ( dispatch ) => {
+		if ( !supportUser || !supportPassword ) {
+			return;
+		}
+
+		dispatch( {
+			type: SUPPORT_USER_FETCH_TOKEN,
+			supportUser
+		} );
+
+		const user = new User();
+
+		const activateSupportUser = ( userData ) => 
+			dispatch( supportUserActivated( Object.assign( {}, user.data ) ) );
+
+		const tokenErrorCallback = ( error ) => 
+			dispatch( supportUserDeactivated( error.message ) );
+
+		const changeUserCallback = error => error
+			? tokenErrorCallback( error )
+			: activateSupportUser();
+
+		user.changeUser( supportUser, supportPassword, changeUserCallback, tokenErrorCallback );
+	}
+}
+/**
+ * Logs out of support user, restoring the original user
+ * @return {object} The action object
+ */
+export function supportUserRestore() {
+	let user = new User();
+	user.clear();
+	user.restoreUser();
```

Comment:
I second what @aduth is saying here, though withheld my comments previously because it seemed to be mostly stage one in moving the legacy code to Redux. But consider this a plus one, a bump, or a thumbs-up to what he said :wink: